app/code/Magento/LayeredNavigation/Block/Navigation.php

```diff
@@ -107,7 +107,8 @@ public function getFilters()
      */
     public function canShowBlock()
     {
-        return $this->visibilityFlag->isEnabled($this->getLayer(), $this->getFilters());
+        return $this->getLayer()->getCurrentCategory()->getDisplayMode() !== \Magento\Catalog\Model\Category::DM_PAGE
```

Comment:
Please fix the styles and ensure lines does not exceed the 120 character limit https://www.php-fig.org/psr/psr-2/#1-overview